### ISSUE TITLE
[dynamo] We are not generating code for x**2 when x is an ndarray

### ISSUE THREAD

**Issue by lezcano** [pytorch/pytorch#109605]
### 🐛 Describe the bug

In this case, we are just falling back to `aten.pow`. This is caused by us casting the 2 into an array.

Tiny repro
```python
import torch
import numpy as np

@torch.compile
def fn(x):
    return x ** 2

fn(np.arange(8))
```

I tagged this as a good first issue, but I don't have time to guide for how to solve it. Fixing this is a matter of debugging through dynamo and finding where are we casting the 2 into a tensor and fixing that. If you want to work on this, simply submit a fix and tag me as a reviewer.

As a side note, it looks like we just do this for int arrays, as for floating point arrays we do generate code.

### Versions

master

cc @mruberry @rgommers @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

**Comment by Gauravkumar1502** [comment]
Hello,
I've  gone through the codebase and identified a specific section that warrants attention in relation to the reported issue.
However, I'd like to confirm if this is indeed the source of the error.
`dtype = next(x.get_dtype() for x in (a, b) if isinstance(x, ir.TensorBox))` inside `lowering.py`

**Comment by lezcano** [comment]
That is getting the dtype of the first of the two variables, but the issue here is that the 2 is being casted to an array, rather than keeping it a scalar.

**Comment by Aashishthakur10** [comment]
@lezcano I've identified the issue, but I'm not sure about how to go about solving it, since it's not specific to Dynamo. The conversion from an `int` to a `tensor` occurs when we attempt to generate a fake tensor for the proxy class.

I noticed that it subsequently calls the `numpy_operator_wrapper`function, further leading to `normalizer(_func=None, *, promote_scalar_result=False)`, somewhere down the chain of NEP50 calls, in function `nep50_to_tensors`, this happens 
```
# finally, can make `weak` into a 0D tensor
weak = to_tensor(weak, dt) <- weak being our integer value 2 converted into a fake tensor! 
```
So, my question is would this require a general change or do we have to control this inside dynamo?

**Comment by lezcano** [comment]
That is exactly the right place to be looking at :)
[NEP 50](https://numpy.org/neps/nep-0050-scalar-promotion.html) will change the rules for how do tensors and scalars and 0-D tensors interact together. In that function we implement that casting.

I get the feeling that in this case, given that the tensor (int64) and the value (pyint) we shouldn't be casting the scalar at all, and we simply should be passing the tensor and the scalar to the given function.
